servers/rendering/rendering_device.cpp

```diff
@@ -188,6 +194,25 @@ void RenderingDevice::shader_set_get_cache_key_function(ShaderSPIRVGetCacheKeyFu
 	get_spirv_cache_key_function = p_function;
 }
 
+void RenderingDevice::register_built_in_include_file(const String &p_filename, const String &p_shader_code) {
+	print_line("Adding build in shader ", p_filename);
+	print_line(p_shader_code);
+
+	build_in_includes[p_filename] = p_shader_code;
+}
+
+bool RenderingDevice::has_built_in_include_file(const String &p_filename) {
+	return build_in_includes.has(p_filename);
+}
+
+String RenderingDevice::get_built_in_include_file(const String &p_filename) {
+	if (build_in_includes.has(p_filename)) {
+		return build_in_includes[p_filename];
+	};
+
+	return String();
```

Comment:
```suggestion
	const String *ptr = build_in_includes.getptr(p_filename);
	
	return ptr ? *ptr : String();
```
Avoids double lookup